scripts/deploy-ranger-service.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###################################
+# Resolve project root directory. #
+###################################
+
+PROJECT_ROOT=$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )")
+
+############################################
+# Resolve Ranger_HOME and  admin password. #
+############################################
+
+source $PROJECT_ROOT/scripts/resolve-ranger-env.sh
+
+if [[ -z $ADMINPASSWD ]] ; then
+	echo "Missing Ranger admin password. Please specify RANGER_HOME environment."
+	exit 0
+fi
+
+#################################
+# Copy jar to ranger class path #
+#################################
+
+MODULE="ranger-gcs-service"
+TARGET_DIR="target"
+SERVICE_DEF_DIR="classes"
+SERVICE_DEF="ranger-servicedef-gcs.json"
+
+PLUGIN_NAME="gcs"
+JAR_DESTINATION="$RANGER_HOME/ranger-admin/ews/webapp/WEB-INF/classes/ranger-plugins/$PLUGIN_NAME"
+
+echo "Start deploying Ranger service."
+echo "Copy jar file to Ranger directory"
+
+# depoly jar
+mkdir -p $JAR_DESTINATION
+JAR_NAME=$(ls $MODULE/$TARGET_DIR | grep jar)
+cp $PROJECT_ROOT/$MODULE/$TARGET_DIR/$JAR_NAME $JAR_DESTINATION/
+
+if [[ $? -ne 0 ]] ; then
+	echo "Something went wrong when copying."
+	exit $?
+fi
+
+chown -R ranger $JAR_DESTINATION/
+chgrp -R ranger $JAR_DESTINATION
+
+##############################################
+# Upload service definition to Ranger server #
+##############################################
+
+echo "Upload service definition to Ranger server."
+
+# ranger settings
+ADMIN="admin:$ADMINPASSWD"
```

Comment:
surround commands that involve password with "set +x" and "set -x" so they don't get accidentally logged.